src/public_types_bindings_mac.rs

```diff
@@ -2,3 +2,33 @@ pub const kPointsPerInch: ::std::os::raw::c_int = 72;
 pub const kMinCredibleResolution: ::std::os::raw::c_int = 70;
 pub const kMaxCredibleResolution: ::std::os::raw::c_int = 2400;
 pub const kResolutionEstimationFactor: ::std::os::raw::c_int = 10;
+
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum PageIteratorLevel {
+    RIL_BLOCK = 0,
+    RIL_PARA = 1,
+    RIL_TEXTLINE = 2,
+    RIL_WORD = 3,
+    RIL_SYMBOL = 4,
+}
+#[repr(u32)]
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum PolyBlockType {
+    PT_UNKNOWN = 0,
+    PT_FLOWING_TEXT = 1,
+    PT_HEADING_TEXT = 2,
+    PT_PULLOUT_TEXT = 3,
+    PT_EQUATION = 4,
+    PT_INLINE_EQUATION = 5,
+    PT_TABLE = 6,
+    PT_VERTICAL_TEXT = 7,
+    PT_CAPTION_TEXT = 8,
+    PT_FLOWING_IMAGE = 9,
+    PT_HEADING_IMAGE = 10,
+    PT_PULLOUT_IMAGE = 11,
+    PT_HORZ_LINE = 12,
+    PT_VERT_LINE = 13,
+    PT_NOISE = 14,
+    PT_COUNT = 15,
+}
```

Comment:
When I run `cargo build` from your branch, my `./target/debug/build/tesseract-sys-261694b8696f235a/out/capi_bindings.rs` looks like this
[public_types_bindings.txt](https://github.com/ccouzens/tesseract-sys/files/11996095/public_types_bindings.txt)

In addition to what's in the branch it contains:

```rust
pub const Orientation_ORIENTATION_PAGE_UP: Orientation = 0;
pub const Orientation_ORIENTATION_PAGE_RIGHT: Orientation = 1;
pub const Orientation_ORIENTATION_PAGE_DOWN: Orientation = 2;
pub const Orientation_ORIENTATION_PAGE_LEFT: Orientation = 3;
pub type Orientation = ::std::os::raw::c_uint;
pub const WritingDirection_WRITING_DIRECTION_LEFT_TO_RIGHT: WritingDirection =
    0;
pub const WritingDirection_WRITING_DIRECTION_RIGHT_TO_LEFT: WritingDirection =
    1;
pub const WritingDirection_WRITING_DIRECTION_TOP_TO_BOTTOM: WritingDirection =
    2;
pub type WritingDirection = ::std::os::raw::c_uint;
pub const TextlineOrder_TEXTLINE_ORDER_LEFT_TO_RIGHT: TextlineOrder = 0;
pub const TextlineOrder_TEXTLINE_ORDER_RIGHT_TO_LEFT: TextlineOrder = 1;
pub const TextlineOrder_TEXTLINE_ORDER_TOP_TO_BOTTOM: TextlineOrder = 2;
pub type TextlineOrder = ::std::os::raw::c_uint;
pub const PageSegMode_PSM_OSD_ONLY: PageSegMode = 0;
pub const PageSegMode_PSM_AUTO_OSD: PageSegMode = 1;
pub const PageSegMode_PSM_AUTO_ONLY: PageSegMode = 2;
pub const PageSegMode_PSM_AUTO: PageSegMode = 3;
pub const PageSegMode_PSM_SINGLE_COLUMN: PageSegMode = 4;
pub const PageSegMode_PSM_SINGLE_BLOCK_VERT_TEXT: PageSegMode = 5;
pub const PageSegMode_PSM_SINGLE_BLOCK: PageSegMode = 6;
pub const PageSegMode_PSM_SINGLE_LINE: PageSegMode = 7;
pub const PageSegMode_PSM_SINGLE_WORD: PageSegMode = 8;
pub const PageSegMode_PSM_CIRCLE_WORD: PageSegMode = 9;
pub const PageSegMode_PSM_SINGLE_CHAR: PageSegMode = 10;
pub const PageSegMode_PSM_SPARSE_TEXT: PageSegMode = 11;
pub const PageSegMode_PSM_SPARSE_TEXT_OSD: PageSegMode = 12;
pub const PageSegMode_PSM_RAW_LINE: PageSegMode = 13;
pub const PageSegMode_PSM_COUNT: PageSegMode = 14;
pub type PageSegMode = ::std::os::raw::c_uint;
```

and
```rust
pub const ParagraphJustification_JUSTIFICATION_UNKNOWN: ParagraphJustification =
    0;
pub const ParagraphJustification_JUSTIFICATION_LEFT: ParagraphJustification = 1;
pub const ParagraphJustification_JUSTIFICATION_CENTER: ParagraphJustification =
    2;
pub const ParagraphJustification_JUSTIFICATION_RIGHT: ParagraphJustification =
    3;
pub type ParagraphJustification = ::std::os::raw::c_uint;
pub const OcrEngineMode_OEM_TESSERACT_ONLY: OcrEngineMode = 0;
pub const OcrEngineMode_OEM_LSTM_ONLY: OcrEngineMode = 1;
pub const OcrEngineMode_OEM_TESSERACT_LSTM_COMBINED: OcrEngineMode = 2;
pub const OcrEngineMode_OEM_DEFAULT: OcrEngineMode = 3;
pub const OcrEngineMode_OEM_COUNT: OcrEngineMode = 4;
pub type OcrEngineMode = ::std::os::raw::c_uint;
```

I'm using a Fedora 37 container.

What happened to them? Did you mean to exclude them, or do they not exist on your system?

I don't mean to sound critical, just trying to work out which of the 3 courses of action to take:
1. Add them in
2. Exclude them using build.rs
3. Leave this file and build.rs as is

Reply:
> What happened to them? Did you mean to exclude them, or do they not exist on your system?

They exist, I meant to exclude them. I only included the bindings I am currently using upstream in the Mac bindings for the reasoning that, as those are the bindings that would break my code if they change.
Those are the only ones I'd know to update if tesseract makes breaking changes to as they are the changes that I am specifically using.

I left the autogenerated bindings for public_types open to generate bindings for any public type not specifically blocklisted. My rationale is that as it currently stands the public_types is overly restrictive about what it provides currently, and as it is "useless" for any project that wants to depend on more than the subset that either the plumbing or -rs crate is currently using. I understand the restriction on leptonica types being leaked, but think that the rest should probably be included.

Please do be critical though, I'm definitely not good at writing code for others to consume. If you think it's dumb that's the only way I'll find out it actually is :)

Reply:
Let's add them in then 🙂

This is a `sys` crate, so some variation between platforms is expected. But I'd like to keep what's supported on Macs roughly the same as what's supported on other platforms.

I can check every now and again if the file needs updating. I may even add a github action to do so.

Reply:
Sounds good to me :)

While I'm at it, I suppose I'll also make them all rustified-enums, unless you have objections to that.